sno/working_copy/postgis.py

```diff
@@ -438,38 +389,59 @@ def write_full(self, commit, *datasets, **kwargs):
 
         with self.session(bulk=2) as db:
             dbcur = db.cursor()
+
+            dbcur.execute(
+                SQL("CREATE SCHEMA IF NOT EXISTS {};").format(Identifier(self.schema))
+            )
+            dbcur.execute(
+                SQL(
+                    """CREATE TABLE IF NOT EXISTS {} (
+                          f_table_catalog    VARCHAR(256) NOT NULL,
+                          f_table_schema     VARCHAR(256) NOT NULL,
+                          f_table_name       VARCHAR(256) NOT NULL,
+                          f_geometry_column  VARCHAR(256) NOT NULL,
+                          coord_dimension    INTEGER NOT NULL,
+                          srid               INTEGER NOT NULL,
+                          type               VARCHAR(30) NOT NULL
+                    );"""
+                ).format(self._table_identifier("geometry_columns"))
```

Comment:
these aren't user-created tables. They're created by Postgis when the extension is installed, and populated by PostGIS:
- `geometry_columns`: this is a _view_ over the column/table definitions, so changes whenever a geometry column is created/altered.
- `spatial_ref_sys` is populated by PostGIS with the EPSG database when the extension is installed, but is user-updatable.

Reply:
Oops! You are absolutely right. Much of this change could be salvaged, but I've made some major alterations. I stopped  creating and writing to either table. Instead I just read from the ones that are already maintained. This works fine unless there is a custom CRS that doesn't match anything in the `spatial_ref_sys` table.

Reply:
change not pushed?

Reply:
Remembered to push, forgot to commit...